src/org/openstreetmap/josm/actions/search/SearchCompiler.java

```diff
@@ -1554,6 +1558,44 @@ public String toString() {
         }
     }
 
+    /**
+     * Matches presets.
+     */
+    private static class Preset extends Match {
+        private List<TaggingPreset> ps;
+
+        Preset(String presetName) throws ParseError {
+
+            if (presetName == null)
+                throw new ParseError("Preset name cannot be null");
```

Comment:
"Preset name is required" (since you use this to validate the parser. If you use it to just check the parameter, use `Objects.requireNonNull`)